### PR TITLE
[ENG-5170] Addon account reauthorization

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-5170]
-   Feature flag: gravy_waffle

## Purpose
- Allow users to reauthorize an addon service account after the credentials have expired

## Summary of Changes
- Update account-setup-manager to handle reauthorization workflow
- Update user-addon-manager and addon settings page to support reauthorization workflow
- Some misc things:
  - Rename `Provider.createAccountForNodeAddon` to `Provider.createAuthorizedAccount` for clarity
  - Move `AuthorizedAccount.configuringUser` relationship to the subclasses as the inverse relationships vary based on service type
  - Allow creation of multiple configuredAddons for a given addon
  - Remove unused `Provider.setNodeAddonCredentials`

## Screenshot(s)
- Desktop
image
image
image
- Mobile
image

image

## QA Notes

[ENG-5170]: https://openscience.atlassian.net/browse/ENG-5170?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mirage/views/addons.ts [hunk 1]
```diff
@@ -315,13 +315,89 @@ export function createAuthorizedComputingAccount(this: HandlerContext, schema: S
...
+        }
+        authorizedAccount.update({
+            ...authorizedAttrs,
+        });
+        return authorizedAccount;
+    } catch (e) {
+        return new Response(403, {}, {
+            errors: [{ detail: e.message }],
+        });
+    }
```
Comment (brianjgeiger) [inline mirage/views/addons.ts:340]: So this try will catch if there's a permissions issue?

**Review by brianjgeiger — APPROVED** [review]
A couple of questions, but nothing blocking.

File: app/settings/addons/template.hbs [hunk 2]
```diff
@@ -54,6 +54,19 @@
                                 @onInput={{manager.onCredentialsInput}}
                             />
                         </div>
+                    {{else if (eq manager.pageMode 'accountReconnect')}}
+                        <div data-analytics-scope='Account Reconnects'>
+                            <h3 local-class='page-heading'>
+                                {{t 'addons.accountSelect.reconnect-account'}}
+                            </h3>
+                            <AddonsService::AccountSetupManager
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply app/settings/addons/template.hbs:62]: Good point. This doesn't manage anything, so I can rename this guy to make it a bit more clear

File: mirage/views/addons.ts (hunk 1 above)
Comment (futa-ikeda) [reply mirage/views/addons.ts:340]: Yea, in `prepareAuthorizedAccountAttrs`, we call `fakeCheckCredentials` that will throw an error if the necessary credentials fields aren't truthy

**Pull request merged**